### DISCUSSION TITLE
Timescale License Dilemma and Burden to Comply

### DISCUSSION THREAD

**Discussion by ozum** [supabase/supabase#28356] [General]
As everybody interested knows, Timescale provides two licenses: a limited version with [Apache 2.0 License](https://github.com/timescale/timescaledb/blob/master/LICENSE-APACHE) and a full-featured version with [Timescale License (TSL)](https://www.timescale.com/legal/licenses). Most valuable features (a.k.a. Community Features) are behind TSL.

Timescale's official licensing page explains:
> Roughly speaking, as long as you are not offering TimescaleDB as a hosted Database-as-a-Service, you can use all Community features for free.

Currently, if you want to use a full-featured Timescale with Supabase, you have ~~two options~~ one option:

1. Self-host Supabase and add Timescale extension yourself.
2. ~~Use the Timescale service and add Supabase yourself. (If possible!)~~ Seems hard/impossible

I want to use the Supabase service (not self-host) and add a full-featured Timescale extension. Is there any way to do this? i.e.

- Is Supabase a database-as-a-service, and does TSL really apply to Supabase?
- Can we manually change our Timescale license to TSL as described [here in Timescale forums](https://www.timescale.com/forum/t/upgrade-your-license-to-timescale-to-use-this-free-community-feature/677). In that case it is not offered by Supabase and seems to comply with licensing terms (or not(?)).

As a last resort, I thought to subscribe to both Supabase and Timescale, divide my database by putting time-related parts into Timescale and the rest in Supabase, and access using a Foreign Data Wrapper or an API. As you guessed, this is not natural micro-service partitioning but architecture forced by legal licensing issues. Such a development burden...

I would appreciate any thoughts or suggestions.

**Comment by GaryAustin1** [comment]
Can't comment on your work a rounds, but Supabase has commented it can't host the extension.  One dev comment here: https://github.com/orgs/supabase/discussions/17108#discussioncomment-8392177

**Comment by servrox** [comment]
Wrote a thread about it + a proper solution! I used a github action to re-build the image.

https://x.com/servrox1337/status/2023679230530211958
